trio/_core/_asyncgens.py

```diff
@@ -22,17 +29,17 @@ class AsyncGenerators:
     # asyncgens after the system nursery has been closed, it's a
     # regular set so we don't have to deal with GC firing at
     # unexpected times.
-    alive = attr.ib(factory=weakref.WeakSet)
+    alive: weakref.WeakSet[AsyncGenerator[Any, Any]] | set[AsyncGenerator[Any, Any]] = attr.ib(factory=weakref.WeakSet)  # type: ignore[assignment]
 
     # This collects async generators that get garbage collected during
     # the one-tick window between the system nursery closing and the
     # init task starting end-of-run asyncgen finalization.
-    trailing_needs_finalize = attr.ib(factory=set)
+    trailing_needs_finalize: set[AsyncGenerator[Any, Any]] = attr.ib(factory=set)
 
     prev_hooks = attr.ib(init=False)
 
-    def install_hooks(self, runner):
-        def firstiter(agen):
+    def install_hooks(self, runner: Runner) -> None:
+        def firstiter(agen: AsyncGenerator[Any, Any]) -> None:
```

Comment:
Instead of `Any`, this could use `AsyncGenerator[object, NoReturn]` - that should accept any generator, but it'll keep it type-safe by preventing code here from actually iterating it and expecting a specific value. Same applies for the rest of the file of course.

Reply:
Changed in 0205c8f